### PR TITLE
Fix bugs

### LINKED ISSUE #655: Different values submitted for the same Mimicry Market mashup
Seeing some odd submissions for one of Mimicry's Market Mashup feeds:

Screenshot 2023-04-13 at 8 37 10 AM

Here's a link for one of the 7 billion-ish submissions: https://polygonscan.com/tx/0x7ea08447cb1ecfba14f64bfefd3a5c4992aa811685ba25d94fe8dd32b477e04c

Here's a link for one of the 5 billion-ish submissions: https://polygonscan.com/tx/0x972fefffd46c5b5ac7c4ee403a27e1577be862139bb567625057193d3cf68759

I don't know which one is the correct value at this time. Will comment. These values are not being used, so we won't be disputing anything or supporting disputers.

I'm picking up the tips for these feeds using the hacked telliot described in issue #652.

### LINKED ISSUE #662: Mimicry Mashup Tip Listener error "InsufficientDataBytes"
I think this error might be because of too many requests on the NFTgo api, but not sure. Full log is attached in the first comment. The mashup feed tip is picked up by the listener, then this error.

start command: `telliot report -a mumbai4 -wp 30`

Versions:
telliot-core #37abbd1d2dd27d5c2278d434d1375d62731c584
telliot-feeds #7f135d927e9e0134cd521d5aadc162befb3f0e56

### LINKED ISSUE #664: Error Message while Listening for Tips on Optimism
It's not clear what's causing this error and it's not a telliot breaking bug. Will provide the log below.

start command: `telliot report -a op3 -pwd "asdf;lkj" -p 50 -wp 60 -tx 0`

versions:
telliot-core #8d117f74ef7a71b1084417ff9c0710625d95a2cd
telliot-feeds #57094013fb068817f7e88e4f7d8c4a283b446235

```
.tellor_360 | Current Oracle stakeAmount: 10.0
INFO    | telliot_feeds.reporters.tellor_360 | 

            STAKER INFO
            start date: 1679962389
            stake_balance: 20.0
            locked_balance: 0
            last report: 1680857004
            reports count: 3
            
ERROR   | telliot_feeds.reporters.tips.listener.funded_feeds | Error getting feeds, timestamps, and values
ERROR   | telliot_feeds.reporters.tips.listener.funded_feeds | Error getting eligible funded feeds: multicall failed to fetch data: ContractLogicError('execution reverted')
INFO    | telliot_feeds.reporters.interval | Unable to suggest datafeed
INFO    | telliot_feeds.reporters.interval | Sleeping for 60 seconds
INFO    | telliot_feeds.reporters.tellor_360 | Current Oracle stakeAmount: 10.0
INFO    | telliot_feeds.reporters.tellor_360 | 

            STAKER INFO
            start date: 1679962389
            stake_balance: 20.0
            locked_balance: 0
            last report: 1680857004
            reports count: 3
            
ERROR   | telliot_feeds.reporters.tips.listener.funded_feeds | Error getting feeds, timestamps, and values
ERROR   | telliot_feeds.reporters.tips.listener.funded_feeds | Error getting eligible funded feeds: multicall failed to fetch data: ContractLogicEr
```

### LINKED ISSUE #665: NFTgo Mimicry Mashup related Error
start command: `telliot report -a polygon1 -p 10 -wp 120`
version: 
telliot-core #037abbd1d2dd27d5c2278d434d1375d62731c584
telliot-feeds #54908ccba1ed65a2cbfd2062418266d661242e76

log:
```
ERROR   | telliot_core | Exception occurred in telliot-core app
ERROR   | telliot_core | <class 'asyncio.exceptions.TimeoutError'>
ERROR   | telliot_core |
ERROR   | telliot_core | ['  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/cli/commands/report.py", line 496, in report\n    await reporter.report()\n', '  File "/home/ubuntu/tell
iot-feeds/src/telliot_feeds/reporters/interval.py", line 511, in report\n    _, _ = await self.report_once()\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/interva
l.py", line 366, in report_once\n    datafeed = await self.fetch_datafeed()\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tellor_360.py", line 282, in fetch_dataf
eed\n    suggested_feed, tip_amount = await get_feed_and_tip(self.autopay)\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/suggest_datafeed.py", line 43, in ge
t_feed_and_tip\n    feed_tips = await funded_feeds.querydata_and_tip(current_time=current_timestamp)\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/f
unded_feeds.py", line 112, in querydata_and_tip\n    eligible_funded_feeds, status = await self.filtered_funded_feeds(\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporte
rs/tips/listener/funded_feeds.py", line 76, in filtered_funded_feeds\n    feeds_timestsamps_and_values_filtered = await self.window_and_priceThreshold_unmet_filter(\n', '  File "/home/
ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py", line 241, in window_and_priceThreshold_unmet_filter\n    price_change = await self.price_change(
\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py", line 112, in price_change\n    value_now = await datafeed.source.fetch_new_d
atapoint()\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 186, in fetch_new_datapoint\n    tokens_mcaps = await self.fetch_tokens_mca
p()\n', '  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 168, in fetch_tokens_mcap\n    responses = await self.fetch_urls(urls=urls)\n', ' 
 File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 67, in fetch_urls\n    responses = await asyncio.gather(*tasks)\n', '  File "/home/ubuntu/te
lliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 85, in fetch_url_with_retry\n    response = await self.fetch(url=url, headers=headers)\n', '  File "/home/ubuntu/t
elliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 52, in fetch\n    async with session.get(url, timeout=timeout) as response:\n', '  File "/home/ubuntu/tenv/lib/py
thon3.9/site-packages/aiohttp/client.py", line 1141, in __aenter__\n    self._resp = await self._coro\n', '  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/client.py", lin
e 560, in _request\n    await resp.start(conn)\n', '  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 914, in start\n    self._continue = None\n', '
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/helpers.py", line 720, in __exit__\n    raise asyncio.TimeoutError from None\n']
Traceback (most recent call last):
  File "/home/ubuntu/tenv/bin/telliot", line 8, in <module>
    sys.exit(main())
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/ubuntu/telliot-core/src/telliot_core/cli/utils.py", line 18, in wrapper
    return asyncio.run(f(*args, **kwargs))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/cli/commands/report.py", line 496, in report
    await reporter.report()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/interval.py", line 511, in report
    _, _ = await self.report_once()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/interval.py", line 366, in report_once
    datafeed = await self.fetch_datafeed()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tellor_360.py", line 282, in fetch_datafeed
    suggested_feed, tip_amount = await get_feed_and_tip(self.autopay)
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/suggest_datafeed.py", line 43, in get_feed_and_tip
    feed_tips = await funded_feeds.querydata_and_tip(current_time=current_timestamp)
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds.py", line 112, in querydata_and_tip
    eligible_funded_feeds, status = await self.filtered_funded_feeds(
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds.py", line 76, in filtered_funded_feeds
    feeds_timestsamps_and_values_filtered = await self.window_and_priceThreshold_unmet_filter(
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py", line 241, in window_and_priceThreshold_unmet_filter
    price_change = await self.price_change(
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py", line 112, in price_change
    value_now = await datafeed.source.fetch_new_datapoint()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 186, in fetch_new_datapoint
    tokens_mcaps = await self.fetch_tokens_mcap()
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 168, in fetch_tokens_mcap
    responses = await self.fetch_urls(urls=urls)
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 67, in fetch_urls
    responses = await asyncio.gather(*tasks)
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 85, in fetch_url_with_retry
    response = await self.fetch(url=url, headers=headers)
  File "/home/ubuntu/telliot-feeds/src/telliot_feeds/sources/mimicry/mashup_source.py", line 52, in fetch
    async with session.get(url, timeout=timeout) as response:
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/client.py", line 560, in _request
    await resp.start(conn)
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/client_reqrep.py", line 914, in start
    self._continue = None
  File "/home/ubuntu/tenv/lib/python3.9/site-packages/aiohttp/helpers.py", line 720, in __exit__
    raise asyncio.TimeoutError from None
asyncio.exceptions.TimeoutError
```

### PR DESCRIPTION
**Pull request by akremstudy**
### Summary

Closes #655 
Code was checking if response is None but now check if falsey and return None so if the response is 0 telliot shouldn't submit.  
Closes #662 
This issue was caused when price threshold check is triggered and there are no values in oracle to compare with so instead now allows feed to be suggested.
Closes #664 
This error was due to a default gas limit that is set in the multicall package but since the tip listener is read only adjusted gas limit to be None to avoid error.
- here is an example transactions to check other chains are not negatively affected, [mumbai-txn](https://mumbai.polygonscan.com/address/0x742bc90878be199cc95bf5690885e5550a8dd315).

Closes #665 
Added a TimeoutError catch in the except block to account for this error type.
### Steps Taken to QA Changes

- Added test for TimeoutError on mashup source
- Added test for tip listener when no before value to compare threshold
- Added test for multicall gas limit 
- Unskipped a test for tip listener
- Ran all the tests

### Checklist

This pull request is:

- [ ] A documentation error, docs update, or typographical error fix
	- No tests or issue needed
- [x] A code fix
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. Fixes without tests will not be accepted unless it's related to the documentation only.
    - Please make sure docs are updated if need be
- [ ] A feature implementation
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure docs updates are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

File: src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py [hunk 1]
```diff
@@ -104,6 +104,16 @@ async def price_change(self, query_data: bytes, value_before: bytes) -> Optional
                 val = getattr(query, param)
                 setattr(datafeed.source, param, val)
 
+        # if no value before short circuit and include feed to be suggested
```
Comment (oraclown) [inline src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py:107]: It's a little unclear to me why returning `10000` when `_get_price_change(previous_val=0, current_val=0))` "includes a feed to be suggested".

This portion of the comment is a little redundant: `if no value before short circuit`, as it's easy to see that it's returning `if not value_before`.

File: src/telliot_feeds/sources/mimicry/mashup_source.py [hunk 2]
```diff
@@ -171,6 +175,9 @@ async def fetch_tokens_mcap(self) -> Optional[Any]:
             if response is not None:
                 try:
                     market_cap = response["market_data"]["market_cap"][self.currency.lower()]
+                    if not market_cap:
+                        logger.debug(f"No value fetched for token: {response['name']}")
```
Comment (oraclown) [inline src/telliot_feeds/sources/mimicry/mashup_source.py:179]: In this case, you're sure `response` will always contain the `name` key?

File: tests/tips/test_feed_suggestion.py [hunk 3]
```diff
@@ -184,3 +185,69 @@ async def test_feed_with_manual_source(autopay_contract_setup, caplog):
     assert datafeed is None
     assert tip is None
     assert "No auto source for feed with query type SpotPrice to check threshold" in caplog.text
+
+
+@pytest.mark.asyncio
+async def test_no_value_before(autopay_contract_setup, caplog):
+    """Test for when there no prior values when checking threshold"""
```
Comment (oraclown) [inline tests/tips/test_feed_suggestion.py:192]: `there no` -> `there are no`

File: tests/tips/test_feed_suggestion.py [hunk 4]
```diff
@@ -184,3 +185,69 @@ async def test_feed_with_manual_source(autopay_contract_setup, caplog):
...
+    feed, tip = await get_feed_and_tip(r.autopay, current_timestamp=chain.time())
+    assert feed is not None
+    assert tip is not None
+    assert " No value before for SpotPrice" in caplog.text
+
+
+@pytest.mark.asyncio
+async def test_low_gas_limit_error(autopay_contract_setup, caplog):
+    """Test ContractLogicError due to low multicall gas limit
+    This error was noticed when trying to report on optimism because of their whacky gas calculations
```
Comment (oraclown) [inline tests/tips/test_feed_suggestion.py:222]: *wacky

Maybe include a link to description of their wacky gas calculations.

**Review by oraclown — APPROVED** [review]
Thanks, @akremstudy ! Great PR summary, fixes, and tests updates. I just left some small suggestions.

File: src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py (hunk 1 above)
Comment (akremstudy) [reply src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py:107]: 10000 represents 100 percent change, will update comment.

File: src/telliot_feeds/sources/mimicry/mashup_source.py (hunk 2 above)
Comment (akremstudy) [reply src/telliot_feeds/sources/mimicry/mashup_source.py:179]: yeah you're right about this.

File: src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py [hunk 5]
```diff
@@ -104,6 +104,16 @@ async def price_change(self, query_data: bytes, value_before: bytes) -> Optional
...
+        if not value_before:
+            logger.debug(f"No value before for {query.type}")
+            return _get_price_change(previous_val=0, current_val=0)
+
+        # before value has to be of length 32 since uint256 has to be 32 bytes to be decoded by eth_abi
+        if len(value_before) != 32:
+            logger.info(f"Before value type isn't uint256 {value_before.hex()}; can't calculate price change")
+            return None
+        # TODO: handle case when value_before is not uint256 but is exactly 32 bytes
```
[the remark this answers is not in the document]
Comment (akremstudy) [reply src/telliot_feeds/reporters/tips/listener/funded_feeds_filter.py:116]: there no specific exception so a simple try except won't work, each byte would probably need to be analyzed for possibly meaning something other than a number.  This should be a rare case anyway and the downside is just an extra submission so probably not a big deal.

**Pull request merged**